server-sdk/src/main/java/com/amazonaws/ml/mms/serversdk/MMSEndpoint.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package com.amazonaws.ml.mms.serversdk;
+
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.FullHttpResponse;
+
+public abstract class MMSEndpoint {
+    public void doGet(FullHttpRequest req, FullHttpResponse res, Context ctx) {}
```

Comment:
FullHttpRequest and FullHttpResponse only work for non-streaming APIs and might not work for trunk encoding or Streaming APIS.

Reply:
Look into adding interfaces for each method.